### ISSUE TITLE
Cannot upload RGB images to OMERO

### ISSUE THREAD

**Issue by ctrueden** [imagej/imagej-omero#55]
```
[ERROR] Exception
java.io.IOException: io.scif.img.ImgIOException: io.scif.FormatException: Error writing to OMERO: imageIndex=0, planeIndex=0
	at io.scif.services.DefaultDatasetIOService.save(DefaultDatasetIOService.java:157) ~[scifio-0.23.0.jar:0.23.0]
	at io.scif.services.DefaultDatasetIOService.save(DefaultDatasetIOService.java:142) ~[scifio-0.23.0.jar:0.23.0]
	at net.imagej.omero.commands.SaveImageToOMERO.run(SaveImageToOMERO.java:70) ~[imagej-omero-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT]
	at org.scijava.command.CommandModule.run(CommandModule.java:201) [scijava-common-2.41.0.jar:2.41.0]
	at org.scijava.module.ModuleRunner.run(ModuleRunner.java:167) [scijava-common-2.41.0.jar:2.41.0]
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:126) [scijava-common-2.41.0.jar:2.41.0]
	at org.scijava.module.ModuleRunner.call(ModuleRunner.java:65) [scijava-common-2.41.0.jar:2.41.0]
	at org.scijava.thread.DefaultThreadService$2.call(DefaultThreadService.java:181) [scijava-common-2.41.0.jar:2.41.0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:262) [na:1.7.0_45]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_45]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_45]
	at java.lang.Thread.run(Thread.java:744) [na:1.7.0_45]
Caused by: io.scif.img.ImgIOException: io.scif.FormatException: Error writing to OMERO: imageIndex=0, planeIndex=0
	at io.scif.img.ImgSaver.writePlanes(ImgSaver.java:688) ~[scifio-0.23.0.jar:0.23.0]
	at io.scif.img.ImgSaver.writeImg(ImgSaver.java:542) ~[scifio-0.23.0.jar:0.23.0]
	at io.scif.img.ImgSaver.writeImg(ImgSaver.java:528) ~[scifio-0.23.0.jar:0.23.0]
	at io.scif.img.ImgSaver.writeImg(ImgSaver.java:497) ~[scifio-0.23.0.jar:0.23.0]
	at io.scif.img.ImgSaver.writeImg(ImgSaver.java:459) ~[scifio-0.23.0.jar:0.23.0]
	at io.scif.img.ImgSaver.saveImg(ImgSaver.java:174) ~[scifio-0.23.0.jar:0.23.0]
	at io.scif.img.ImgSaver.saveImg(ImgSaver.java:158) ~[scifio-0.23.0.jar:0.23.0]
	at io.scif.services.DefaultDatasetIOService.save(DefaultDatasetIOService.java:154) ~[scifio-0.23.0.jar:0.23.0]
	... 11 common frames omitted
Caused by: io.scif.FormatException: Error writing to OMERO: imageIndex=0, planeIndex=0
	at net.imagej.omero.OMEROFormat.writerException(OMEROFormat.java:663) ~[imagej-omero-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT]
	at net.imagej.omero.OMEROFormat.access$600(OMEROFormat.java:77) ~[imagej-omero-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT]
	at net.imagej.omero.OMEROFormat$Writer.writePlane(OMEROFormat.java:517) ~[imagej-omero-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT]
	at io.scif.AbstractWriter.savePlane(AbstractWriter.java:168) ~[scifio-0.23.0.jar:0.23.0]
	at io.scif.AbstractWriter.savePlane(AbstractWriter.java:159) ~[scifio-0.23.0.jar:0.23.0]
	at io.scif.img.ImgSaver.writePlanes(ImgSaver.java:685) ~[scifio-0.23.0.jar:0.23.0]
	... 18 common frames omitted
Caused by: omero.ResourceError: null
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:1.7.0_45]
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57) ~[na:1.7.0_45]
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:1.7.0_45]
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526) ~[na:1.7.0_45]
	at java.lang.Class.newInstance(Class.java:374) ~[na:1.7.0_45]
	at IceInternal.BasicStream.createUserException(BasicStream.java:2615) ~[omero-client-5.1.1-ice35.jar:na]
	at IceInternal.BasicStream.access$300(BasicStream.java:12) ~[omero-client-5.1.1-ice35.jar:na]
	at IceInternal.BasicStream$EncapsDecoder10.throwException(BasicStream.java:3099) ~[omero-client-5.1.1-ice35.jar:na]
	at IceInternal.BasicStream.throwException(BasicStream.java:2077) ~[omero-client-5.1.1-ice35.jar:na]
	at IceInternal.Outgoing.throwUserException(Outgoing.java:538) ~[omero-client-5.1.1-ice35.jar:na]
	at omero.api._RawPixelsStoreDelM.setPlane(_RawPixelsStoreDelM.java:1668) ~[omero-client-5.1.1-ice35.jar:na]
	at omero.api.RawPixelsStorePrxHelper.setPlane(RawPixelsStorePrxHelper.java:8082) ~[omero-client-5.1.1-ice35.jar:na]
	at omero.api.RawPixelsStorePrxHelper.setPlane(RawPixelsStorePrxHelper.java:8044) ~[omero-client-5.1.1-ice35.jar:na]
	at net.imagej.omero.OMEROFormat$Writer.writePlane(OMEROFormat.java:514) ~[imagej-omero-0.3.0-SNAPSHOT.jar:0.3.0-SNAPSHOT]
	... 21 common frames omitted
```
Issue #54 may or may not be related.

**Comment by ctrueden** [comment]
Confirmed that this error still happens with an OMERO 5.1.1 server and client library combination. Stack trace updated accordingly.

**Comment by joshmoore** [comment]
@ctrueden : could you also copy/upload/etc the server log?

**Comment by ctrueden** [comment]
> could you also copy/upload/etc the server log?

Good idea. I will post it as soon as I have access to the relevant hard drive again (I foolishly left it at work last night, and am home sick today).

My current (weak) theory is that there is probably some dimension order scrambling going on: see [OMEROFormat.java#L352-L357](https://github.com/imagej/imagej-omero/blob/cdf0b77b579b15f039e3955c3712dd1512bef025/src/main/java/net/imagej/omero/OMEROFormat.java#L352-L357) vs. [OMEROFormat.java#L572-L573](https://github.com/imagej/imagej-omero/blob/cdf0b77b579b15f039e3955c3712dd1512bef025/src/main/java/net/imagej/omero/OMEROFormat.java#L572-L573). But even if so, you'd think the planes would simply be out of order, rather than the upload blowing up.

**Comment by ctrueden** [comment]
Well, with the addition of 4f779f4e8731609d21a8606640e51f240c1b2a32, this problem is resolved! Uploading 5D data (including Mitosis sample) as well as RGB data (including Clown sample) works now.

Summary of embarrassing events:

1.  Identified the problem.

2.  Refactored the codebase to facilitate a cleaner bug-fix (e4401c45de04c9de9eb8e002d398b8f73ee8325a).

3.  Made the bug-fix (b3be0529a7de9f137d10c2bc7ae8466e281bbf7d).

4.  Tested bug-fix and saw that surprisingly, it didn't fix the issue.

5.  Wrung hands and pulled hair.

6.  Thanks to @joshmoore's suggestion, noticed the following amusing Blitz log entry:

        Blitz-0.log:2015-05-18 16:31:31,676 INFO  [        ome.services.util.ServiceHandler] (.Server-79)  Args:    [320, 320, 320, 320, (0, 1, 2, ... 317 more), ome.model.enums.PixelsType:Id_5, clown.jpg, clown.jpg]

7. Realized the prior refactoring actually introduced a second bug. Fixed it (4f779f4e8731609d21a8606640e51f240c1b2a32).

8. Retested the original bug-fix, now rebased (fac85beeba4e5c177c133a808480df3bfdbe4894), and found the issue was indeed actually resolved.

9. \*headdesk\*
